worldedit-core/src/main/java/com/sk89q/worldedit/function/pattern/BiomePattern.java

```diff
@@ -31,7 +32,33 @@
      * Return a {@link BiomeType} for the given position.
      *
      * @param position the position
-     * @return a block
+     * @return a biome
+     * @deprecated use {@link BiomePattern#applyBiome(BlockVector3)}
      */
-    BiomeType apply(BlockVector2 position);
+    @Deprecated
+    default BiomeType apply(BlockVector2 position) {
+        return applyBiome(position.toBlockVector3());
+    }
+
+    /**
+     * Return a {@link BiomeType} for the given position.
+     *
+     * <p>
+     *     Note: This will not be default in WE8.
+     * </p>
+     *
+     * @param position the position
+     * @return a biome
+     */
+    default BiomeType applyBiome(BlockVector3 position) {
+        // TODO Remove default in WE8
+        try {
+            if (getClass().getMethod("apply", BlockVector2.class).getDeclaringClass().equals(BiomePattern.class)) {
```

Comment:
At this point I'm wondering if extracting this to a utility is worthwile. Something like `DeprecationUtil`:
```java
public static void checkDelegatingOverride(Class<?> declaringClass, Class<?> implementingClass, String newMethodName, String oldMethodName, Class<?>... parameterTypes) {
  try {
    Method method = implementingClass.getMethod(oldMethodName, parameterTypes);
    if (method.getDeclaringClass().equals(declaringClass)) {
      throw new IllegalStateException("Class " + implementingClass.getName() + " must override " + newMethodName);
    }
  } catch (NoSuchMethodException e) {
    throw new AssertionError(e);
  }
}
```

Usage:
```java
DeprecationUtil.checkDelegatingOverride(getClass(), BiomePattern.class, "applyBiome(BlockVector3)", "apply", BlockVector2.class);
```

Might be nicer to do some sort of stack walking to more easily acquire the new method name, but that's the general idea.